packages/catalyst/src/dashboard/components/_shared/Form/Form.tsx

```diff
@@ -53,6 +54,8 @@ export const Form: React.FC<Props> = ({
   const onSubmit = form.handleSubmit(async data => {
```

Comment:
Shouldn't we create a custom hook to handle all this logic? It'd return `pending` and `setPending` and also take a `successMessage` and an `errorMessage` to trigger the toast 
Maybe it could be useful in other future situations

Reply:
I don't like how complex such a hook would be. It would need to take `method`, `endpoint`, `locale`, `form`, `errorMessage` and `successMessage`, and return `data`, `errors`, `pending`, `setPending` etc. If we could re-use it somewhere I think it might be worth it, but for now I think we can't.

That said, that block of code is still 😬. If we manage to use server actions it'd get a lot simpler, as the number of dependencies would be reduced by at least 3. It would become a simple try/catch that shows a toast, and we can use something like [zact](https://github.com/pingdotgg/zact) for the data-fetching logic.